src/core/lwt_seq.ml

```diff
@@ -0,0 +1,131 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                 Simon Cruanes                                          *)
+(*                                                                        *)
+(*   Copyright 2017 Institut National de Recherche en Informatique et     *)
+(*     en Automatique.                                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
```

Comment:
Even though nobody will bother with this realistically, we can't have a part of the code claiming to be distributed under a (modified) LGPL license in an otherwise MIT-licensed Lwt, which doesn't (and shouldn't) make an exception for it &mdash; especially if it sends people to a `LICENSE` file that isn't present in Lwt.

I think the detailed authorship of this code needs to be addressed. My (uninformed) guess is that this code was originally written by @c-cube, and modified by @zshipko for Lwt. So, we should ask @c-cube for permission to distribute this variant of it in Lwt under the MIT license, in the same way as all the other Lwt code. In particular, we don't include the names of authors in each file, but rely on blame for detailed authorship history.

If @c-cube agrees, @zshipko, please change these license headers to match the ones in the rest of Lwt. If this code is indeed substantially @c-cube's, we can also express that using a Co-authored-by field in the commit, and including a note about where the code comes form in the commit's detailed message. @raphael-proust can then make sure that is preserved during the squash.

Reply:
You can probably just write all that in the message of a future commit in this PR, probably thet same one that would change the license headers, and, again, @raphael-proust can move the Co-authored-by and the history message to the right place when squashing.

Reply:
I'll make sure that the squashed commit includes the relevant info from the commit message.

Reply:
You can use it with the MIT license as far as I'm concerned, and there's no need to list me as an author. If you start adapting large portions of [oseq](https://github.com/c-cube/oseq/) (which you should :wink:) you can also port it with MIT but list me as a coauthor.

Reply:
Thanks, @c-cube!